src/preview/topics.rs

```diff
@@ -0,0 +1,292 @@
+use momento_protos::cache_client::pubsub::{
+    self, pubsub_client::PubsubClient, PublishRequest, SubscriptionRequest,
+};
+use tonic::{codegen::InterceptedService, transport::Channel};
+
+use crate::MomentoError;
+use crate::{
+    endpoint_resolver::MomentoEndpointsResolver,
+    grpc::header_interceptor::HeaderInterceptor,
+    utils::{connect_channel_lazily, user_agent},
+    MomentoResult,
+};
+
+type ChannelType = InterceptedService<Channel, HeaderInterceptor>;
+
+pub struct Pubsub {
+    client: pubsub::pubsub_client::PubsubClient<ChannelType>,
+}
+
+/// Work with topics, publishing and subscribing.
+impl Pubsub {
```

Comment:
Just a naming question / quirk here.  Are we calling this feature Pubsub?  Or Topics?  Just want to try to make our branding consistent.

Also I think if we decide to go with PubSub as the outer name then the S should be capitalized.

Reply:
I don't know that we actually have a decision on this. If we want to capitalize the S I'll just go TopicClient instead :-). I don't like the way PubSub reads. I'll update with TopicClient thanks.